src/OpenTelemetry/Sdk.cs

```diff
@@ -41,13 +42,33 @@ static Sdk()
             Activity.DefaultIdFormat = ActivityIdFormat.W3C;
             Activity.ForceDefaultIdFormat = true;
             SelfDiagnostics.EnsureInitialized();
+
+            var assemblyInformationalVersion = typeof(Sdk).Assembly.GetCustomAttribute<AssemblyInformationalVersionAttribute>()?.InformationalVersion;
+            if (string.IsNullOrWhiteSpace(assemblyInformationalVersion))
+            {
+                assemblyInformationalVersion = "1.0.0";
+            }
+
+            /*
+             * InformationalVersion will be in the following format:
+             *   {majorVersion}.{minorVersion}.{patchVersion}.{pre-release label}.{pre-release version}.{gitHeight}+{Git SHA of current commit}
+             * Ex: 1.5.0-alpha.1.40+807f703e1b4d9874a92bd86d9f2d4ebe5b5d52e4
+             * The following parts are optional: pre-release label, pre-release version, git height, Git SHA of current commit
+             */
+
+            var indexOfPlusSign = assemblyInformationalVersion!.IndexOf('+');
+            InformationalVersion = indexOfPlusSign > 0
+                ? assemblyInformationalVersion.Substring(0, indexOfPlusSign)
+                : assemblyInformationalVersion;
         }
 
         /// <summary>
         /// Gets a value indicating whether instrumentation is suppressed (disabled).
         /// </summary>
         public static bool SuppressInstrumentation => SuppressInstrumentationScope.IsSuppressed;
 
+        internal static string InformationalVersion { get; }
```

Comment:
```suggestion
        internal static readonly string InformationalVersion;
```

Reply:
I'm not opposed to this, but why? 🤣 A property with just a getter (`{ get; }`) is `readonly` for the backing generated by the compiler IIRC.

Reply:
> but why? 

Just because it's more obvious when you use `readonly` 😀